### PR TITLE
Start Using BeginSignIn command and remove obsolete API command calls

### PR DESCRIPTION
**Pull request by yuwenmemon**
@robertjchen please review

### Details
Start using the `BeginSignIn` command added in https://github.com/Expensify/Web-Expensify/pull/34408

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/218743

### Tests
1. Do a normal sign-in flow for a user that already has a validated account, make sure it works properly:
https://user-images.githubusercontent.com/4741899/182488142-8044224b-396a-46b2-97a2-69599c0ee76d.mp4

---

1. With the same account do the same sign in. Click "Forgot?" on the password form.
2. Make sure you see the following:
Screen Shot 2022-08-02 at 4 03 52 PM

3. And that you get send a password reset link email.
4. Click the "Resend link" button.
5. Make sure you get another email

---

1. Sign in with an email that DOES NOT already have an account
2. Make sure that right away you see:
Screen Shot 2022-08-02 at 4 14 01 PM

3. ...And that you get an email with a sign-in link:
Screen Shot 2022-08-02 at 4 14 25 PM

4. Before validating that account by clicking the link in the email, sign-in with that account again, make sure that you get an email with a sign-in link again.

---

1. Sign-in with a phone number that does not have an account yet, make sure you see the following:
Screen Shot 2022-08-02 at 4 16 39 PM

2. And that you get a text message with your magic sign-in link:
IMG_8397

---

### CODE REVIEW

File: src/libs/actions/Session/index.js [hunk 1]
```diff
@@ -138,44 +103,41 @@ function resendValidationLink(login = credentials.login) {
  * @param {String} login
  */
 function fetchAccountDetails(login) {
```
Comment (luacmartins) [inline src/libs/actions/Session/index.js:105]: Should we rename this to `beginSignIn` to match the API command?

File: README.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ For an M1 Mac, read this [SO](https://stackoverflow.com/c/expensify/questions/11
 1. If you are having issues with **_Getting Started_**, please reference [React Native's Documentation](https://reactnative.dev/docs/environment-setup)
```
Comment (luacmartins) [inline README.md:65]: We should update the other examples using `ONYXKEYS.ACCOUNT` to `isLoading` instead of `loading`

**Review by luacmartins — CHANGES_REQUESTED** [review]
Left a few comments. In addition, we have a few places in [actions/User.js](https://github.com/Expensify/App/blob/238465d109e6ddd6fea31644994db969563a2319/src/libs/actions/User.js#L53) that still set `loading` in `ONYXKEYS.ACCOUNT`. We should update those.

File: src/libs/actions/Session/index.js (hunk 1 above)
Comment (robertjchen) [reply src/libs/actions/Session/index.js:105]: +1

File: src/pages/SetPasswordPage.js [hunk 3]
```diff
@@ -104,15 +91,15 @@ class SetPasswordPage extends Component {
...
+        if (this.props.credentials.authToken) {
+            Session.changePasswordAndSignIn(this.props.credentials.authToken, this.state.password);
         } else {
             Session.setPassword(this.state.password, validateCode, accountID);
         }
     }
 
     render() {
-        const buttonText = this.props.userSignUp.isValidating ? this.props.translate('setPasswordPage.verifyingAccount') : this.props.translate('setPasswordPage.setPassword');
+        const buttonText = !this.props.account.validated ? this.props.translate('setPasswordPage.verifyingAccount') : this.props.translate('setPasswordPage.setPassword');
```
Comment (robertjchen) [inline src/pages/SetPasswordPage.js:102]: Is it possible to consolidate the remaining usage of `verifyingAccount` to `validatingAccount`? 🤔  It seems like we're using both `validating` and `verifying` interchangeably, with the former appearing to be the more common occurrence.

**Review by robertjchen — CHANGES_REQUESTED** [review]

**Comment by yuwenmemon** [comment]
Trying to figure out how the hell I broke 2 unit tests that don't seem related at all 🤔

**Comment by yuwenmemon** [comment]
Okay updated!

**Review by luacmartins — COMMENTED** [review]
Thanks for addressing the comments! Tests are still failing though!

**Comment by yuwenmemon** [comment]
They are not! 😄

**Review by robertjchen — DISMISSED** [review]
Awesome, thanks!!

File: src/pages/SetPasswordPage.js [hunk 4]
```diff
@@ -133,7 +120,7 @@ class SetPasswordPage extends Component {
                     <View>
                         <FormAlertWithSubmitButton
                             buttonText={buttonText}
-                            isLoading={this.props.account.loading || this.props.userSignUp.isValidatingEmail}
+                            isLoading={this.props.account.loading}
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/pages/SetPasswordPage.js:123]: Do we need to update this one?

**Review by luacmartins — CHANGES_REQUESTED** [review]

File: src/pages/SetPasswordPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (yuwenmemon) [reply src/pages/SetPasswordPage.js:123]: Yes good catch!

**Comment by yuwenmemon** [comment]
Updated again!

**Review by luacmartins — APPROVED** [review]

**Pull request merged**